frontend/src/Frontend/UI/Transfer.hs

```diff
@@ -1037,40 +1041,57 @@ gasPayersSection model netInfo fks ti = do
         pure $ (dgp1, Nothing)
       else do
         let mkLabel c = T.pack $ printf "Gas Paying Account (Chain %s)" (T.unpack $ _chainId c)
+            destinationStatus = Map.lookup toAccount tks
+            defaultDestGasPayer = case destinationStatus of
+              Just (AccountStatus_Exists _) -> toAccount
+              _ -> AccountName "free-x-chain-gas"
         (_,dgp1) <- uiAccountNameInput (mkLabel fromChain) True (Just $ _ca_account $ _ti_fromAccount ti) never noValidation
-        (_,dgp2) <- uiAccountNameInput (mkLabel toChain) True Nothing never noValidation
+        (_,dgp2) <- uiAccountNameInput (mkLabel toChain) True (Just defaultDestGasPayer) never noValidation
         pure $ (dgp1, Just dgp2)
-    let getGasPayerKeys chain mgp = do
-          case mgp of
-            Nothing -> return never
-            Just gp -> do
-              -- I think lookupKeySets can't be done in the PushM monad
-              evt <- lookupKeySets (model ^. logger) (_sharedNetInfo_network netInfo)
-                            (_sharedNetInfo_nodes netInfo) chain [gp]
-              return $ Map.lookup gp <$> fmapMaybe id evt
+    let
+      getGasPayerKeys chain mgp = do
+        case mgp of
+          Nothing -> return never
+          Just gp -> do
+            -- I think lookupKeySets can't be done in the PushM monad
+            evt <- lookupKeySets (model ^. logger) (_sharedNetInfo_network netInfo)
+                          (_sharedNetInfo_nodes netInfo) chain [gp]
+            return $ Map.lookup gp <$> fmapMaybe id evt
 
     -- Event t (Maybe AccountName)
     gp1Debounced <- debounce 1.0 $ updated dgp1
-    -- Maybe (Event t (Maybe AccountName))
-    mgp2Debounced <- maybe (return Nothing) (fmap Just . debounce 1.0 . updated) mdgp2
-    -- also not sure what to do if the source chain gas payer is the same as dest chain gas payer
-    -- should we let the user choose different signers for the source and dest chain?
     gp1Keys <- networkHold (return never)
                            (getGasPayerKeys fromChain <$> gp1Debounced)
     let initialDetails = Map.lookup fromAccount fks
     gp1 <- foldDyn ($) (GasPayerDetails fromChain (Just fromAccount) initialDetails) $ leftmost
       [ (\a gp -> gp { gpdAccount = a }) <$> updated dgp1
       , (\keys gp -> gp { gpdDetails = keys }) <$> switch (current gp1Keys)
       ]
-    case mgp2Debounced of
-      Nothing -> return $ GasPayers gp1 Nothing
-      Just dgp2 -> do
-        gp2Keys <- networkHold (return never) (getGasPayerKeys toChain <$> dgp2)
+    --Outer maybe represents single chain transfer (so no second gas payer). Inner maybe represents
+    --an account name that is less than 3 chars for the "Destination Gas Payer" input field
+    mgp2 <- forM mdmgp2 $ \dmgp2 -> do
+        -- Take the initial value of the destination gas payer and lookup its details
+        initVal <- sample $ current dmgp2
```

Comment:
What's the reason for using `sample` here?  I have a general heuristic that `sample` is often a bad idea.  It carries with it the big question mark of when the sample occurs and is usually more strict than you actually want.  Using `networkHold` or `networkView` tend to make it a lot more clear what event is triggering the action.

Reply:
Switched to just use the initial default value for the input-field